cpp/src/community/edge_triangle_count_impl.cuh

```diff
@@ -78,77 +78,298 @@ struct update_edges_p_r_q_r_num_triangles {
+         pair_dsts     = pair_dsts.data(),
+         pair_count    = (*pair_count).data(),
+         edge_first] __device__(auto idx) {
+          auto src          = pair_srcs[idx];
+          auto dst          = pair_dsts[idx];
+          auto p_r_q_r_pair = thrust::make_tuple(src, dst);
+
+          // Find its position in 'edges'
+          auto itr_p_r_q_r =
+            thrust::lower_bound(thrust::seq, edge_first, edge_first + num_edges, p_r_q_r_pair);
+
+          assert(*itr_p_r_q_r == p_r_q_r_pair);
+          auto idx_p_r_q_r = thrust::distance(edge_first, itr_p_r_q_r);
+
+          cuda::atomic_ref<edge_t, cuda::thread_scope_device> atomic_counter(
+            num_triangles[idx_p_r_q_r]);
+          auto r = atomic_counter.fetch_add(pair_count[idx], cuda::std::memory_order_relaxed);
+        });
+
+    } else {
+      // Given intersection offsets and indices that are used to update the number of
+      // triangles of (p, q) edges where `r`s are the intersection indices, update
+      // the number of triangles of the pairs (p, r) and (q, r).
+      thrust::for_each(
+        handle.get_thrust_policy(),
+        thrust::make_counting_iterator<edge_t>(0),
+        thrust::make_counting_iterator<edge_t>(intersection_indices.size()),
+        update_edges_p_r_q_r_num_triangles<vertex_t, edge_t, decltype(edge_first)>{
+          edgelist_srcs.size(),
+          0,
+          prev_chunk_size,
+          raft::device_span<size_t const>(intersection_offsets.data(), intersection_offsets.size()),
+          raft::device_span<vertex_t const>(intersection_indices.data(),
+                                            intersection_indices.size()),
+          raft::device_span<edge_t>(num_triangles.data(), num_triangles.size()),
+          edge_first});
+
+      thrust::for_each(
+        handle.get_thrust_policy(),
+        thrust::make_counting_iterator<edge_t>(0),
+        thrust::make_counting_iterator<edge_t>(intersection_indices.size()),
+        update_edges_p_r_q_r_num_triangles<vertex_t, edge_t, decltype(edge_first)>{
+          edgelist_srcs.size(),
+          1,
+          prev_chunk_size,
+          raft::device_span<size_t const>(intersection_offsets.data(), intersection_offsets.size()),
+          raft::device_span<vertex_t const>(intersection_indices.data(),
+                                            intersection_indices.size()),
+          raft::device_span<edge_t>(num_triangles.data(), num_triangles.size()),
+          edge_first});
+    }
+    prev_chunk_size += chunk_size;
+  }
+
+  cugraph::edge_property_t<graph_view_t<vertex_t, edge_t, false, multi_gpu>, edge_t> counts(
+    handle, graph_view);
+
+  cugraph::edge_bucket_t<vertex_t, void, true, multi_gpu, true> valid_edges(handle);
+  valid_edges.insert(edgelist_srcs.begin(), edgelist_srcs.end(), edgelist_dsts.begin());
+
+  auto cur_graph_view = graph_view;
+
+  auto edge_last = edge_first + edgelist_srcs.size();  // FIXME: Remove this unnecessary variable
```

Comment:
You can make this update in the capture list (i.e. line 346) to minimize the scope of this temporary..